services/py-api/src/server/handlers/participants_handlers.py

```diff
@@ -21,7 +22,12 @@ async def create_participant(
         #     result = await self._service.register_admin_participant(input_data)
         # else:
         #     ...
-        result = await self._service.register_admin_participant(input_data)
+
+
+        if input_data.is_admin and input_data.team_name:
+            result = await self._service.register_admin_participant(input_data)
+        else:   
+            result = await self._service.register_random_participant(input_data)
```

Comment:
This is logically incorrect. If a participant is not an admin, but it has a team_name it will still be considered a random participant (although a random participant does not have a team). It should be something like: 

```Python
if input_data.is_admin:
#register the participant as admin
else if input_data.team_name:
#the participant wants to join a team through an invite link
else
#the participant is joining as a random participant
```

Reply:
Fixed in 988821a7cdf6e6755270c0cae57e8480e6456db2, so I am resolving this.